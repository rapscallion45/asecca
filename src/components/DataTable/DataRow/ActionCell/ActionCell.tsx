import { FC, ReactNode, memo } from 'react';
import StyledTableCell from '../StyledCellWrapper/StyledCellWrapper';

/**
 * Data Table Action Cell Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.13
 *
 * @typedef IActionCellProps
 * @prop {ReactNode} getActionComponent - get component to present
 * @prop {any} sx - cell styling overrrides
 */
interface IActionCellProps {
  getActionComponent?: () => ReactNode;
  sx?: any;
}

/**
 * Data Table Action Cell
 *
 * Action table cell component for displaying action to user
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.13
 *
 * @component
 * @param {IActionCellProps} props - component props
 * @returns {FC} - data table action cell functional component
 */
const ActionCell: FC<IActionCellProps> = (props) => {
  const { getActionComponent, sx } = props;

  return (
    <StyledTableCell align="left" sx={sx}>
      {getActionComponent && getActionComponent()}
    </StyledTableCell>
  );
};

export default memo(ActionCell);
