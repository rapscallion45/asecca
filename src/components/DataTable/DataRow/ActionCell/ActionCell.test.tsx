import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Table, TableBody, TableRow } from '@mui/material';
import ActionCell from './ActionCell';

/* test text */
const testValue = 'Test Component';

/* test action component */
const getTestComponent = () => <div>{testValue}</div>;

/**
 * Data Table Action Cell Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.22
 */
describe('Data Table Action Cell', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = renderer
      .create(
        <Table>
          <TableBody>
            <TableRow>
              <ActionCell getActionComponent={getTestComponent} />
            </TableRow>
          </TableBody>
        </Table>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Value', () => {
    it('Should render passed component', async () => {
      /** Arrange */
      /** Act */
      render(
        <Table>
          <TableBody>
            <TableRow>
              <ActionCell getActionComponent={getTestComponent} />
            </TableRow>
          </TableBody>
        </Table>
      );

      /** Assert */
      expect(screen.getByText(testValue)).toBeInTheDocument();
    });
  });
});
