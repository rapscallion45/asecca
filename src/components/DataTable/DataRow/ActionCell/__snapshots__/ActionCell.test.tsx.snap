// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Action Cell Renders correctly 1`] = `
<table
  className="MuiTable-root css-rqglhn-MuiTable-root"
  role={null}
>
  <tbody
    className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
      >
        <div>
          Test Component
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
